src/views/Visualization.vue

```diff
@@ -109,6 +123,28 @@ export default {
           })
         })
       },
+      lazyLoadImages(){
+        const loadImg = function(entries, observer){
+          const [entry] = entries;
+          //Guard Clause
+          //If image is not intersecting viewport do nothing
+          if (!entry.isIntersecting) return;
+          //Get first source element
+          entry.target.srcset = entry.target.dataset.srcset; 
+          //Get second source element
+          entry.target.nextElementSibling.srcset = entry.target.dataset.srcset; 
+          //stop observing img after the switch
+          observer.unobserve(entry.target);
+        }
+        const imgTargets = document.querySelectorAll(".lazy > source");
+        const imgObserver = new IntersectionObserver(loadImg, {
+          //Watch entire viewport
+          root: null,
+          threshold: 0,
+          rootMargin: "300px"
```

Comment:
So `rootMargin` is where the trigger is located? Is this in view or off the page?

Reply:
Yep, so the threshold 0 is the very top of the observed element and I am saying 300px before that load in the image.

Reply:
I had it set to 200px initially but mobile works better with a larger margin.